File: VB/WindowsFormsApplication3/InplaceDataNavigator/InplaceDataNavigatorPainter.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms

Namespace WindowsFormsApplication3
	Public Class InplaceDataNavigatorPainter
		Inherits BaseEditPainter
		Public Overrides Sub Draw(ByVal info As ControlGraphicsInfoArgs)
			MyBase.Draw(info)
			Dim vi As BaseEditViewInfo = TryCast(info.ViewInfo, BaseEditViewInfo)
			Dim cri As RepositoryItemInplaceDataNavigator = TryCast(vi.Item, RepositoryItemInplaceDataNavigator)
			Dim e As New PaintEventArgs(info.Graphics, vi.Bounds)
			cri.EditNavigator.Draw(e)
		End Sub
	End Class
End Namespace

File: CS/WindowsFormsApplication3/InplaceDataNavigator/InplaceDataNavigatorPainter.cs
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;

namespace WindowsFormsApplication3
{
    public class InplaceDataNavigatorPainter : BaseEditPainter
    {
        public override void Draw(ControlGraphicsInfoArgs info)
        {
            base.Draw(info);
            BaseEditViewInfo vi = info.ViewInfo as BaseEditViewInfo;
            RepositoryItemInplaceDataNavigator cri = vi.Item as RepositoryItemInplaceDataNavigator;
            PaintEventArgs e = new PaintEventArgs(info.Graphics, vi.Bounds);      
            cri.EditNavigator.Draw(e);            
        }
    }
}
